packages/cli/src/commands/deploy.ts

```diff
@@ -191,6 +195,28 @@ const deployPlan = async (
   return result
 }
 
+
+type GroupWithArtifacts = Group & Required<Pick<Group, 'accountName' | 'artifacts'>>
+const isArtifactsGroup = (group: Group): group is GroupWithArtifacts => (
+  group.artifacts !== undefined && group.accountName !== undefined
+)
+
+const writeArtifacts = (
+  groups: types.NonEmptyArray<GroupWithArtifacts>,
+  artifactsDir: string,
+): void => {
+  log.debug('writing deploy artifacts to %s', artifactsDir)
+  groups.forEach(({ accountName, artifacts }) => {
+    const artifactDir = `${artifactsDir}/${accountName ?? 'unknown'}`
```

Comment:
isn't `accountName` a required string here?
```suggestion
    const artifactDir = `${artifactsDir}/${accountName}`
```

Reply:
While it never should be undefined here, this is a tech debt due to the fact the adapter & core share the `DeployResult` interface. See my other comment on the interface itself 🙏

Reply:
My bad, literally forgot I added this typeguard there 😆 👍